### PR TITLE
Allow user to set config_file via environment variable

### LINKED ISSUE #194: Allow user to set config_file via environment variable
Currently, the default value for the `config_file` argument is `"~.slackr"`
```r
slackr_setup(
  channel = "#general",
  username = "slackr",
  icon_emoji = "",
  incoming_webhook_url = "",
  token = "",
  config_file = "~/.slackr",
  echo = FALSE,
  cache_dir = ""
) 
```
and if a user need to use another path, they have to set it in every function call explicitly e.g.
```r
slackr_setup(config_file = "/etc/slackr")
```
however, this could be easily solved by setting an environment variable in `.Renviron` and passing it while keeping the default value the same like below
```r
slackr_setup(
  channel = "#general",
  username = "slackr",
  icon_emoji = "",
  incoming_webhook_url = "",
  token = "",
  config_file = Sys.getenv("SLACKR_CONFIG_FILE", unset = "~/.slackr"),
  echo = FALSE,
  cache_dir = ""
) 
```
Happy to make a PR if you think that's okay.

### PR DESCRIPTION
**Pull request by botan**
Closes #194

### CODE REVIEW

File: R/slackr_setup.r [hunk 1]
```diff
@@ -55,7 +55,7 @@ slackr_setup <- function(
   icon_emoji = "",
   incoming_webhook_url = "",
   token = "",
-  config_file = "~/.slackr",
+  config_file = Sys.getenv("SLACKR_CONFIG_FILE", unset = "~/.slackr"),
```
Comment (mrkaye97) [inline R/slackr_setup.r]: do you mind changing this to `SLACKR_CONFIG_FILE_PATH`?

File: README.Rmd [hunk 2]
```diff
@@ -86,10 +86,12 @@ The vignettes contain setup instructions and example usage:
 
 ### Config File Setup
 
-The `slackr_setup()` function will try to read setup values from a `~/.slackr`
-(you can change the default) configuration file, which may be easier and more
-secure than passing them in manually (plus, will allow you to have multiple
-`slackr` configurations for multiple Slack.com teams).
+The `slackr_setup()` function will try to read setup values from a `~/.slackr` 
+(you can change the default filepath by recording in the SLACKR_CONFIG_FILE
```
Comment (mrkaye97) [inline README.Rmd]: corresponding change here for var name

File: README.md [hunk 3]
```diff
@@ -87,10 +87,11 @@ roughly the same, with only slightly differing scopes.
 ### Config File Setup
 
 The `slackr_setup()` function will try to read setup values from a
-`~/.slackr` (you can change the default) configuration file, which may
-be easier and more secure than passing them in manually (plus, will
-allow you to have multiple `slackr` configurations for multiple
-Slack.com teams).
+`~/.slackr` (you can change the default filepath by recording in the
+SLACKR_CONFIG_FILE environment variable or supplying as an argument to
```
Comment (mrkaye97) [inline README.md]: corresponding change

**Review by mrkaye97 — COMMENTED** [review]
Thanks for putting this in! Looks like I might need to wrestle with CI a little bit -- I'll report back

**Comment by botan** [comment]
No problem, please let me know if anything else I might need to change.

**Comment by mrkaye97** [comment]
merging into dev. thanks for the contribution! I'll keep wrestling with CI to get this over the line

**Comment by botan** [comment]
Cheers!

**Pull request merged**